examples/cache/src/main/java/momento/client/example/LoadGenerator.java

```diff
@@ -0,0 +1,262 @@
+
+  private final int requestsPerSecond;
+
+  private final ConcurrentHistogram setHistogram = new ConcurrentHistogram(3);
+  private final ConcurrentHistogram getHistogram = new ConcurrentHistogram(3);
+  private final LongAdder globalRequestCount = new LongAdder();
+  private final LongAdder globalSuccessCount = new LongAdder();
+  private final LongAdder globalUnavailableCount = new LongAdder();
+  private final LongAdder globalTimeoutCount = new LongAdder();
+  private final LongAdder globalLimitExceededCount = new LongAdder();
+
+  private final String cacheValue;
+  private final CacheClient client;
+
+  private final long startTime;
+
+  public LoadGenerator(
+      int statsInterval, int maxConcurrentRequests, int requestsPerSecond, int cacheValueLength) {
+    cacheValue = "x".repeat(cacheValueLength);
+
+    final CredentialProvider credentialProvider;
+    try {
+      credentialProvider = new EnvVarCredentialProvider(AUTH_TOKEN_ENV_VAR);
+    } catch (SdkException e) {
+      logger.error("Unable to load credential from environment variable " + AUTH_TOKEN_ENV_VAR, e);
+      throw e;
+    }
+    client =
+        CacheClient.builder(credentialProvider, Configurations.Laptop.v1(), DEFAULT_ITEM_TTL)
+            .build();
+
+    client.createCache(CACHE_NAME);
+
+    executorService = Executors.newScheduledThreadPool(maxConcurrentRequests);
+    rateLimiter = RateLimiter.create(requestsPerSecond);
+    this.requestsPerSecond = requestsPerSecond;
+
+    startTime = System.currentTimeMillis();
+
+    // Schedule initial tasks
+    for (int i = 0; i < maxConcurrentRequests; ++i) {
+      scheduleSet(i, 0);
+    }
+
+    // Schedule a task to print the stats
+    statsScheduler.scheduleAtFixedRate(
+        this::logInfo, statsInterval, statsInterval, TimeUnit.SECONDS);
+  }
+
+  private void scheduleSet(int workerId, int operationNum) {
+    scheduleOperation(
+        workerId,
+        operationNum,
+        key -> client.set(CACHE_NAME, key, cacheValue),
+        (response, operationNumValue) -> {
+          if (response instanceof SetResponse.Success) {
+            globalSuccessCount.increment();
+          } else if (response instanceof SetResponse.Error error) {
+            handleErrorResponse(error.getErrorCode());
+          }
+          scheduleGet(workerId, operationNumValue);
+        },
+        setHistogram);
+  }
+
+  private void scheduleGet(int workerId, int operationNum) {
+    final int nextOperationNum = operationNum + 1;
+    scheduleOperation(
+        workerId,
+        operationNum,
+        key -> client.get(CACHE_NAME, key),
+        (response, operationNumValue) -> {
+          if (response instanceof GetResponse.Hit || response instanceof GetResponse.Miss) {
+            globalSuccessCount.increment();
+          } else if (response instanceof GetResponse.Error error) {
+            handleErrorResponse(error.getErrorCode());
+          }
+          scheduleSet(workerId, nextOperationNum);
```

Comment:
This looks like unintended else we'll be in an infinite loop of set-get-set?

Reply:
That is intended. It schedules alternating write and read tasks that reschedule themselves forever. The throughput is controlled by the rate limiter blocking when the tasks run too quickly. It is broken by the shutdown method that the main method calls at the end of the benchmark.